novice-goals.Rmd

```diff
@@ -0,0 +1,226 @@
+
+## Data manipulation
+
+-   How can I read tabular data into a program?
+    -   what CSV is, where it comes from, and why people use it
+    -   reading files
+-   How can I select subsets of my data?
+    -   select
+    -   filter
+    -   arrange
+    -   Boolean conditions
+-   How can I calculate new values?
+    -   mutate
+    -   ifelse
+-   How can I tell what's gone wrong in my programs?
+    -   reading error messages
+    -   the difference between syntax and runtime errors
+-   How can I operate on subsets of my data?
+    -   group
+    -   summarize
+    -   split-apply-combine
+-   How can I work with two or more datasets?
+    -   join
+-   How can I save my results?
+    -   writing files
+-   What *isn't* included?
+    -   anything other than reasonably tidy tabular data
+    -   map
+    -   loops and conditionals
+
+## Plotting
+
+-   Why plot?
+    -   summary statistics can mislead
+    -   [Anscombe's Quartet and the DataSaurus dozen][anscome-datasaurus]
+-   What are the core elements of every plot?
+    -   data
+    -   geometric objects
+    -   aesthetic mapping
+-   How can I create different kinds of plots?
+    -   scatter plot
+    -   line plot
+    -   histogram
+    -   bar plot
+    -   which to use when
+-   How can I plot multiple datasets at once?
+    -   grouping
+    -   faceting
+-   How can I make misleading plots?
+    -   showing a single central tendency data point instead of the individual observations
+    -   saturated plots instead of for example violins or 2D histograms
+    -   not starting axis at zero to exaggerate a difference
+    -   not using perceptually uniform colormaps to indicate quantities
+    -   not thinking about color blindness
+-   What *isn't* included?
+    -   outliers
+    -   interactive plots
+    -   maps
+    -   3D visualization
+
+## Development
+
+-   How can I make my own functions?
+    -   declaring functions
+    -   declaring parameters
+    -   default values
+    -   common conventions
+-   How can I make my programs tell me that something has gone wrong?
+    -   validation (did we build the right thing) vs. verification (did we build the thing correctly)
+    -   assertions for sanity checks
+-   How can I ask for help?
+    -   creating a reproducible example (reprex)
+-   How do I install software?
+    -   what *is* a package?
+    -   package manager
+-   What *isn't* included?
+    -   code browsers, multiple cursors, and other fancy IDE tricks
+    -   virtual environments
+    -   debuggers
+
+## Data analysis
+
+-   How can I represent and manage missing values?
+    -   NA
+-   How can I get a feel for my data?
+    -   summary statistics
+-   How can I create a simple model of my data?
+    -   formulas
+    -   linear regression
+        -   adding a best fit straight line on a scatterplot
+        -   understanding what the error bands on a "best fit" straight line mean
+    -   k-means cluster analysis
```

Comment:
Any particular reason for showing k-means? Not sure how commonly used it is in other fields. I know most biomedical researchers don't ever use it. Maybe just discuss generally about clustering, mention a few methods, then focus on k-means?

Reply:
I think kmeans is a good algorithm to introduce early because it is quite simple to understand the basic concepts of how it works. It also often performs well enough to at least be a good starting point when doing clustering/partitioning, even on high dimensional data. I have actually seen it used in many papers in my corner of the biomedical research field.

Reply:
What @joelostblom said: k-means is (in my experience) the simplest kind of clustering to teach.